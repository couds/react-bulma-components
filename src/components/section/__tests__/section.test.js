import React from 'react';
import { render } from 'react-testing-library';
import { Section } from '..';

describe('Section component', () => {
  it('should render', () => {
    const { asFragment } = render(<Section />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should concat classname in props with classname', () => {
    const { asFragment } = render(
      <Section className="other-class this-is-a-test">
        <p>Default</p>
      </Section>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should use inline styles', () => {
    const { asFragment } = render(
      <Section style={{ height: 250 }}>
        <p>Default</p>
      </Section>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should be Large', () => {
    const { asFragment } = render(
      <Section size="large">
        <p>Default</p>
      </Section>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render as nav element', () => {
    const { asFragment } = render(
      <Section renderAs="nav">
        <p>Default</p>
      </Section>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
