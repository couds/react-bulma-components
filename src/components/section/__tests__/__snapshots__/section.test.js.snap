// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component should be Large 1`] = `
<DocumentFragment>
  <section
    class="section is-large"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Section component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <section
    class="section other-class this-is-a-test"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Section component should render 1`] = `
<DocumentFragment>
  <section
    class="section"
  />
</DocumentFragment>
`;

exports[`Section component should render as nav element 1`] = `
<DocumentFragment>
  <nav
    class="section"
  >
    <p>
      Default
    </p>
  </nav>
</DocumentFragment>
`;

exports[`Section component should use inline styles 1`] = `
<DocumentFragment>
  <section
    class="section"
    style="height: 250px;"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;
